source/helpers/conditionParser.ts

```diff
@@ -0,0 +1,119 @@
+export const evaluateAnswer = (
+  questionId: string,
+  answers: Record<string, any>,
+  questions: Question[]
+): boolean => {
+  const question = questions.find((q) => q.id === questionId);
+  if (!question) return false;
+
+  switch (question.type) {
+    case 'checkbox':
+      if (answers[questionId]) return answers[questionId];
+      return false;
+    case 'text':
+    case 'number':
+    case 'date':
+      return (answers[questionId] && answers[questionId] !== '');
+    case 'editableList':
+      return (
+        answers[questionId] &&
+        Object.keys(answers[questionId]).filter(
+          (key) => answers[questionId][key] && answers[questionId][key] !== ''
+        ).length > 0
+      );
+    case 'repeaterField':
+      return (answers[questionId] && answers[questionId].length > 0);
+    case 'summaryList':
+      return evaluateSummaryList(answers, question.items || []);
+    default:
+      return false;
+  }
+};
+
+type EvaluatedValue = boolean | '!' | '&&' | '||';
+
+// evaluation functions
+const evaluateNot = (array: EvaluatedValue[]) => {
+  const arrCopy = [...array];
+  const reversedIndex = arrCopy.reverse().findIndex(expr => expr === '!');
+  if (reversedIndex === -1) return array;
+  
+  const index = arrCopy.length - reversedIndex - 1;
+  arrCopy.reverse()[index+1] = !arrCopy[index+1];
+  arrCopy.splice(index,1);
+  return evaluateNot(arrCopy);
+};
+const evaluateAnd = (array: EvaluatedValue[]) => {
+  const arrCopy = [...array];
+  const index = arrCopy.findIndex(expr => expr === '&&');
+  if (index === -1) return array;
+  
+  arrCopy[index-1] = (array[index-1] && array[index+1]) as EvaluatedValue;
+  arrCopy.splice(index,2);
+  return evaluateAnd(arrCopy);
+}
+const evaluateOr = (array: EvaluatedValue[]): boolean[] => {
+  const arrCopy = [...array];
+  const index = arrCopy.findIndex(expr => expr === '||');
+  if (index === -1) return (array as boolean[]);
+  
+  arrCopy[index-1] = (array[index-1] || array[index+1]) as EvaluatedValue;
+  arrCopy.splice(index,2);
+  return evaluateOr(arrCopy);
+}
+
+const specialWords = ['!', '&&', '||'];
+// regex matching the special words above
+const regex = /(!|&&|\|\|)/gm;
+
+/**
+ * Evaluates an expression of the form "questionId1 && questionId2 || !questionId3".
+ * Allowed boolean operators are !, &&, ||. Parenthesis are not supported.
+ * Evaluates ! first, then && and lastly ||.
+ * @param condition the conditional expression
+ * @param answers all answers of the form
+ * @param questions array of all the questions in the form
+ */
+export const parseConditionalExpression = (
+  condition: string,  
+  answers: Record<string, any>,
+  questions?: Question[]): boolean => {
+    if (!Array.isArray(questions)) return false;
+
+    const conditionArray = condition.split(regex).map(string => string.trim()).filter(string => string !== '');
+    //evaluate all question-ids: 
+    const evaluatedArray: EvaluatedValue[] = conditionArray.map(expression => {
```

Comment:
Renaming conditionArray to conditionAsArray, expression to valueOrOperator.

Reply:
conditionAsArray - njää... conditions räcker

Reply:
but it's not multiple conditions, it's literally one conditon, but represented as an array. So, I don't like the name conditions.

Reply:
varför måste du berätta att det är en array? skapa variabel med värde i från array'en. det skapar onödig förvirring att hämta något i en lista som bara ska innehålla ett enda värde.

försök sträva efter tydlighet även för en som inte normalt läser kod varje dag.

Reply:
No, the array has multiple values; it's just that all of it represents a single condition. For example, the conditon is 'x && y', then the array is ['x', '&&', 'y']. If I call it conditions, then it signifies that there's more than one condition, which is wrong.

Reply:
okej, tror jag fattar vad du gör nu. är det alltid 3 värden i listan?
Kanske något i denna stil skulle lätta upp det för mig
`[termLeft, operator, termRight] = condition`

Säkert feltolkat. Mycket magi här... :)

Reply:
No, the array can have arbitrarily many values, since you can enter any kind of boolean expression, like x && ! y || z, etc. So it's really an array.